src/backend/distributed/planner/distributed_planner.c

```diff
@@ -287,6 +302,60 @@ distributed_planner(Query *parse,
 }
 
 
+/*
+ * ErrorIfQueryHasMergeCommand walks over the query tree and throws error
+ * if there are any Merge command (e.g., CMD_MERGE) in the query tree.
+ */
+static void
+ErrorIfQueryHasMergeCommand(Query *queryTree)
+{
+	/*
+	 * Postgres currently doesn't support Merge queries inside subqueries and
+	 * ctes, but lets be defensive and do query tree walk anyway.
+	 *
+	 * We do not call this path for fast-path queries to avoid this additional
+	 * overhead.
+	 */
+	if (ContainsMergeCommandWalker((Node *) queryTree))
+	{
+		ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+						 errmsg("MERGE command is not supported on Citus tables yet")));
+	}
+}
+
+
+/*
+ * ContainsMergeCommandWalker walks over the node and finds if there are any
+ * Merge command (e.g., CMD_MERGE) in the node.
+ */
+static bool
+ContainsMergeCommandWalker(Node *node)
+{
+#if PG_VERSION_NUM >= PG_VERSION_15
```

Comment:
NIT: #if PG_VERSION_NUM < 14
return false;
and let the rest of the code fall through.

Reply:
Same with above, `CMD_MERGE` has to be wrapped within if statement otherwise build fails in pg13/14.